src/pytest/test_followhashes.py

```diff
@@ -487,6 +487,49 @@ def testEvicted(env):
     env.assertLess(res[0], 1000)
     env.assertGreater(res[0], 0)
 
+def createExpire(env, N):
+  env.cmd('FLUSHALL')
+  env.expect('FT.CREATE idx SCHEMA txt1 TEXT n NUMERIC').ok()
+  for i in range(N):
+    env.expect('HSET', 'doc%d' % i, 'txt1', 'hello%i' % i, 'n', i)
+    env.expect('PEXPIRE doc%d 100' % i)
+  env.expect('HSET', 'foo', 'txt1', 'hello', 'n', 0).equal(2)
+  env.expect('HSET', 'bar', 'txt1', 'hello', 'n', 20).equal(2)
+  waitForIndex(env, 'idx')
+  env.expect('FT.SEARCH', 'idx', 'hello*', 'limit', '0', '0').noEqual([2L])
+  env.expect('HGETALL doc99').equal(['txt1', 'hello99', 'n', '99'])
+  sleep(0.1)
+  env.expect('HGETALL doc99').equal([])
+
+def testExpiredDuringSearch(env):
+  N = 100
+  createExpire(env, N)
+  res = env.cmd('FT.SEARCH', 'idx', 'hello*', 'nocontent', 'limit', '0', '200')
+  env.assertGreater(103, len(res))
```

Comment:
@ashtul not follow how you check you do not have empty results?

Reply:
In the case of nocontent we don't access the key and therefore it isn't expired. 
We get a random number of documents redis has actively expired before the query.

Reply:
@MeirShpilraien in the next one we don't have nocontent and we have only the two documents we didn't set expire on.